pxr/base/tf/unicodeUtils.h

```diff
@@ -164,38 +239,49 @@ class TfUtf8CodePointIterator final {
     /// Advances the iterator logically one UTF-8 character sequence in
     /// the string. The underlying string iterator will be advanced
     /// according to the variable length encoding of the next UTF-8
-    /// character. Invalid leading bytes will increment until the end
-    /// of the range is reached.
+    /// character, but will never consume non-continuation bytes after
+    /// the current one.
     TfUtf8CodePointIterator& operator++ ()
     {
+        // The increment operator should never be called if it's past
+        // the end. The user is expected to have already checked this
+        // condition.
+        TF_DEV_AXIOM(!_IsPastTheEnd());
+        _EncodingLength increment = _GetEncodingLength();
         // note that in cases where the encoding is invalid, we move to the
         // next byte this is necessary because otherwise the iterator would
```

Comment:
Nit-pick that I realize is in a part of the PR that wasn't changed (and also isn't part of a documentation comment): Maybe break into separate sentences or otherwise add punctuation to separate things --> "we move to the next byte. This is necessary..."